test/DGmethods/advection_diffusion/pseudo1D_advection_diffusion.jl

```diff
@@ -175,63 +179,109 @@ let
+  expected_result[2, 3, Float32, VerticalDirection] = 5.3235460654832423e-05
+  expected_result[2, 4, Float32, VerticalDirection] = 3.8958251025178470e-06
+  expected_result[3, 1, Float32, VerticalDirection] = 6.6147454082965851e-02
+  expected_result[3, 2, Float32, VerticalDirection] = 5.7507432065904140e-03
+  expected_result[3, 3, Float32, VerticalDirection] = 8.3523096691351384e-05
+  expected_result[3, 4, Float32, VerticalDirection] = 2.3923123080749065e-04
 
   numlevels = integration_testing ? 4 : 1
 
   @testset "$(@__FILE__)" for ArrayType in ArrayTypes
     for FT in (Float64, Float32)
       result = zeros(FT, numlevels)
       for dim = 2:3
-        n = SVector(1, 1, dim == 2 ? 0 : 1) / FT(sqrt(dim))
-        α = FT(1)
-        β = FT(1 // 100)
-        μ = FT(-1 // 2)
-        δ = FT(1 // 10)
-        for l = 1:numlevels
-          Ne = 2^(l-1) * base_num_elem
-          brickrange = ntuple(j->range(FT(-1); length=Ne+1, stop=1), dim)
-          periodicity = ntuple(j->false, dim)
-          topl = BrickTopology(mpicomm, brickrange; periodicity = periodicity)
-          @show dt = (α/4) / (Ne * polynomialorder^2)
-
-          timeend = 1
-          outputtime = 1
-
-          dt = outputtime / ceil(Int64, outputtime / dt)
-
-          @info (ArrayType, FT, dim)
-          vtkdir = "vtk_advection" *
-          "_poly$(polynomialorder)" *
-          "_dim$(dim)_$(ArrayType)_$(FT)" *
-          "_level$(l)"
-          result[l] = run(mpicomm, ArrayType, dim, topl, polynomialorder,
-                          timeend, FT, dt, n, α, β, μ, δ, vtkdir,
-                          outputtime)
-          @test result[l] ≈ FT(expected_result[dim, l, FT])
-        end
-        @info begin
-          msg = ""
-          for l = 1:numlevels-1
-            rate = log2(result[l]) - log2(result[l+1])
-            msg *= @sprintf("\n  rate for level %d = %e\n", l, rate)
+        for direction in (EveryDirection, HorizontalDirection,
+                          VerticalDirection)
+          if direction <: EveryDirection
+            n = dim == 2 ? SVector{3, FT}(1/sqrt(2), 1/sqrt(2), 0) :
+                           SVector{3, FT}(1/sqrt(3), 1/sqrt(3), 1/sqrt(3))
+          elseif direction <: HorizontalDirection
+            n = dim == 2 ? SVector{3, FT}(1, 0, 0) :
+                           SVector{3, FT}(1/sqrt(2), 1/sqrt(2), 0)
+          elseif direction <: VerticalDirection
+            n = dim == 2 ? SVector{3, FT}(0, 1, 0) : SVector{3, FT}(0, 0, 1)
+          end
+          α = FT(1)
+          β = FT(1 // 100)
+          μ = FT(-1 // 2)
+          δ = FT(1 // 10)
+          for l = 1:numlevels
+            Ne = 2^(l-1) * base_num_elem
+            brickrange = ntuple(j->range(FT(-1); length=Ne+1, stop=1), dim)
+            periodicity = ntuple(j->false, dim)
+            topl = StackedBrickTopology(mpicomm, brickrange;
+                                        periodicity = periodicity)
+            @info dt = (α/4) / (Ne * polynomialorder^2)
```

Comment:
@mwarusz pointed out that this is evaluated in its own local scope (unlike `@show`)
```suggestion
            dt = (α/4) / (Ne * polynomialorder^2)
            @info "time step" dt
```